### ISSUE TITLE
Email Forwarding

### ISSUE THREAD

**Issue by reesericci** [obl-ong/admin#13]
This is a tracking issue for Email Forwarding, an upcoming feature.

We plan to allow for people to forward domain@obl.ong to one or many emails, most likely using DNSimple.

**Comment by reesericci** [comment]
We now have an email server using mailcow

**Comment by sonicx180** [comment]
Hows this going?

**Comment by youngchief-btw** [comment]
We need to create a separate Gem for this feature to keep it away from the core code AFAIK. The email server is working and it's externally accessible (as in, it's exposed to the internet right now). The planned format is [domain]@obl.ong. So for example, if you have trash.obl.ong, then you can get email forwarded for trash@obl.ong. 

- Not sure what we plan to do in terms of directly on your subdomain such as [anything]@trash.obl.ong. 

I can see this being very handy over services like ImprovMX or even paying your email provider for forwarding.

- Something that might be nice to have is allow people to send from their Obl.ong email using SMTP (for example, scripting). 

This could enable a variety of new projects to implement something using email. Though I hope people don't try and use it for high-volume sending or anything too crazy. (Think huge forums vs small personal automation)

- I don't think we want to implement actual inboxes. 

It takes up significantly more resources, and email providers like Proton and Skiff suit this use case better

One problem I see with those last two is a way higher chance for abuse, but we'll have to see what the team decides on going forward. Again, just my thoughts, not speaking for the team as a whole. Hopefully we can get this done soon!

**Comment by sonicx180** [comment]
Alright, thanks!

**Comment by reesericci** [comment]
> We need to create a separate Gem for this feature to keep it away from the core code AFAIK

Yeah, I think that's best - need to learn how to properly make plugin systems though, this is also relevant to the implementation of #15.

> Not sure what we plan to do in terms of directly on your subdomain such as [anything]@trash.obl.ong.

My plan anyway was to do nothing and allow people to use their domain with g suite, etc.

> The email server is working and it's externally accessible (as in, it's exposed to the internet right now).

False - all we have is postmark for transactional mail. The mailcow server is borked atm.

> I don't think we want to implement actual inboxes.

Yeah, I could see this being a paid feature (we're nonprofit - so this would go towards service operations & labor potentially, never profits) down the line but not right now.

I think the best route to implement this is using Action Mailer & Mailbox to have it tightly integrated with the Rails application for email processing -- I think that would be preferable to relying on an external mail server. Action Mailbox and Mailer could connect to an external SMTP/IMAP server but all the mail processing would be handled by Rails.

**Comment by reesericci** [comment]
https://forum.obl.ong/t/model-relations-policies-architecture/44?u=reesericci
